common/dal2/src/ops/io.rs

```diff
@@ -37,3 +94,24 @@ impl futures::AsyncRead for CallbackReader {
         r
     }
 }
+
+#[cfg(test)]
+mod test {
+    use futures::io::Cursor;
+    use futures::StreamExt;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn reader_stream() {
+        let reader = Box::new(Cursor::new("Hello, world!"));
+        let mut s = ReaderStream::new(reader);
+
+        let mut bs = Vec::new();
+        while let Some(chunk) = s.next().await {
+            bs.extend_from_slice(&chunk.unwrap());
+        }
+
+        assert_eq!(&bs[..], "Hello, world!".as_bytes());
```

Comment:
@dantengsky, Can you take a look of this PR please?

The test can't passed for `ReaderStream` returns `None` instead of `Some(Bytes)`. I don't know what I did wrong.

Thanks!

Reply:
Please see https://github.com/datafuselabs/databend/pull/3821/files/a032f4599ec5be1ac87594634d15f3d020fea52d#r781677912